vcsh

```diff
@@ -234,17 +235,20 @@ foreach() {
 	# We default to prefixing `git` to all commands passed to foreach, but
 	# allow running in general context with -g
 	command_prefix=git
-	while getopts "g" flag; do
+	command_suffix='cat'
+	while getopts "gp" flag; do
 		if [ x"$1" = x'-g' ]; then
 			unset command_prefix
 		fi
+		if [ x"$1" = x'-p' ]; then
+			command_suffix='sed "s/^/$VCSH_REPO_NAME: /"'
+		fi
 		shift 1
 	done
 	for VCSH_REPO_NAME in $(list); do
-		echo "$VCSH_REPO_NAME:"
 		GIT_DIR=$VCSH_REPO_D/$VCSH_REPO_NAME.git; export GIT_DIR
 		use
-		$command_prefix "$@"
+		$command_prefix "$@" | eval $command_suffix
```

Comment:
This doesn't feel right to me, I think we can probably do better with an `exec` call *before* running the command to setup the output stream processing. That would avoid piping everything through `cat` even when this option is not active, hence not potentially breaking existing possibilities.